bootloader/grub.go

```diff
@@ -483,3 +483,59 @@ func (g *grub) TrustedAssets() ([]string, error) {
 		"EFI/boot/grubx64.efi",
 	}, nil
 }
+
+// RecoveryBootChain returns the chain of recovery mode boot files for the
+// specified recovery system and kernel. This function should be called
+// only for the recovery bootloader.
+func (g *grub) RecoveryBootChain(kernelPath string) ([]BootFile, error) {
+	if !g.recovery {
+		return nil, fmt.Errorf("not a recovery bootloader")
+	}
+
+	recoveryTrustedAssets, err := g.TrustedAssets()
+	if err != nil {
+		return nil, err
+	}
+
+	// add trusted assets to the recovery chain
+	chain := make([]BootFile, 0, len(recoveryTrustedAssets)+1)
+	for _, ta := range recoveryTrustedAssets {
+		chain = append(chain, NewBootFile("", ta, RoleRecovery))
+	}
+	// add recovery kernel to the recovery chain
+	chain = append(chain, NewBootFile(kernelPath, "kernel.efi", RoleRecovery))
+
+	return chain, nil
+}
+
+// BootChain returns the chain of run mode boot files for the specified run
+// mode bootloader and kernel. This function should be called only for the
+// recovery bootloader.
+func (g *grub) BootChain(runBl TrustedAssetsBootloader, kernelPath string) ([]BootFile, error) {
+	if !g.recovery {
+		return nil, fmt.Errorf("not a recovery bootloader")
+	}
+
+	recoveryTrustedAssets, err := g.TrustedAssets()
+	if err != nil {
+		return nil, err
+	}
+
+	trustedAssets, err := runBl.TrustedAssets()
```

Comment:
this will not work nicely because it requires a NoBootSlash set on runBl which seems a bit of too strict requirement for here.

Reply:
but once we know that runBl.Name is grub we could cast runBl to *grub and then use an internal method of our choice